.github/workflows/automatic-doc-checks.yml

```diff
@@ -1,16 +1,24 @@
-name: Main Documentation Checks
+#
+name: Automatic doc checks
 
 on:
-  - push
-  - pull_request
-  - workflow_dispatch
+  push:
+    branches: [ main ]
+  pull_request:
+    paths:
+      - '.**'   # Only run on changes to the docs directory
+
+  workflow_dispatch:
+    # Manual trigger
+
 
 concurrency:
   group: ${{ github.workflow }}-${{ github.ref }}
-  cancel-in-progress: true  
+  cancel-in-progress: true
 
 jobs:
   documentation-checks:
     uses: canonical/documentation-workflows/.github/workflows/documentation-checks.yaml@main
     with:
-      working-directory: '.'
+      working-directory: "."
+      fetch-depth: 0
```

Comment:
Delete these lines, not needed.

Reply:
Removing this results in the automatic check not starting. While I think those checks are often very flaky, it also feels wrong to disable them entirely, especially the linkcheck which can in fact catch issues (e.g., intersphinx links that no longer work, or such). I've reinstated them.